core/domain/suggestion_services.py

```diff
@@ -255,9 +258,36 @@ def reject_suggestion(suggestion, reviewer_id, review_message):
     mark_review_completed(
         suggestion, suggestion_models.STATUS_REJECTED, reviewer_id)
 
+    thread_id = suggestion.suggestion_id
+    if not constants.ENABLE_GENERALIZED_FEEDBACK_THREADS:
+        thread_id = thread_id[thread_id.find('.') + 1:]
     feedback_services.create_message(
-        suggestion.suggestion_id, reviewer_id,
-        feedback_models.STATUS_CHOICES_IGNORED, None, review_message)
+        thread_id, reviewer_id, feedback_models.STATUS_CHOICES_IGNORED,
+        None, review_message)
+
+
+def get_all_suggestions_that_can_be_reviewed_by_user(user_id):
+    """Returns a list of suggestions which need to be reviewed, in categories
+    where the user has crossed the minimum score to review.
+
+    Args:
+        user_id: str. The ID of the user.
+
+    Returns:
+        list(Suggestion). A list of suggestions which the given user is allowed
+            to review.
+    """
+    score_categories = (
+        user_models.UserContributionScoringModel
+        .get_all_categories_where_user_can_review(user_id))
+
+    if len(score_categories) == 0:
+        return []
+
+    return (
+        suggestion_models.GeneralSuggestionModel
+        .get_in_review_suggestions_in_score_categories(
```

Comment:
Could user get his own suggestions in the UI?

Reply:
Hmmm, it's possible. However, he will not be able to accept/reject them. It will error out saying "cannot accept/reject your own suggestion"

Reply:
Can we filter based on the user id?

Reply:
Done